dev-scheme/racket/racket-8.2.ebuild

```diff
@@ -0,0 +1,172 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit desktop pax-utils xdg-utils
+
+DESCRIPTION="General purpose, multi-paradigm Lisp-Scheme programming language"
+HOMEPAGE="https://racket-lang.org/"
+SRC_URI="
+	minimal? ( https://download.racket-lang.org/installers/${PV}/${PN}-minimal-${PV}-src-builtpkgs.tgz )
+	!minimal? ( https://download.racket-lang.org/installers/${PV}/${P}-src-builtpkgs.tgz )
+"
+S="${WORKDIR}/${P}/src"
+
+# See https://blog.racket-lang.org/2019/11/completing-racket-s-relicensing-effort.html
+LICENSE="
+	|| ( MIT Apache-2.0 )
+	chez? ( Apache-2.0 )
+	!chez? ( LGPL-3 )
+"
+# Bytecode generated by Racket is not compatible between versions (do not include revisions)
+SLOT="0/${PV}"
+KEYWORDS="~amd64"
+IUSE="X +chez +doc +futures +jit minimal +places +readline +threads"
+REQUIRED_USE="futures? ( jit )"
+
+DEPEND="
+	!dev-tex/slatex
+	dev-db/sqlite:3
+	dev-libs/libffi:=
+	X? (
+		dev-util/desktop-file-utils
+		media-libs/libpng:0
+		virtual/jpeg:0
+		x11-libs/cairo[X]
+		x11-libs/gtk+:3[X]
+		x11-libs/pango[X]
+		x11-misc/shared-mime-info
+	)
+	readline? ( dev-libs/libedit )
+"
+RDEPEND="${DEPEND}"
+
+# BC library is being stripped unconditionally
+QA_PRESTRIPPED="/usr/lib64/libracket3m-${PV}.so"
+
+post_X_update() {
```

Comment:
Actually, let's just do:
```
use X && xdg_pkg_postinst
```
and so on in pkg_postinst (and its friend) and drop this?

(Inherit xdg.eclass instead).

Reply:
> xdg.eclass 

then we get unwanted deps: dev-util/desktop-file-utils & x11-misc/shared-mime-info

Reply:
I do wish the second point in this would go through :/
https://archives.gentoo.org/gentoo-dev/message/44462535c5d5e5194994ff5a24d0fb6c

But guess the push for IDEPEND is distracting from this option.

On a side-note, with xdg.eclass `use X &&` wouldn't be needed, it's designed to do nothing if there's no .desktop and no icons needing cache updates.

Reply:
Oh, of course. But yes, the dep :|

Reply:
> I do wish the second point in this would go through :/
> https://archives.gentoo.org/gentoo-dev/message/44462535c5d5e5194994ff5a24d0fb6c

this would be nice


> On a side-note, with xdg.eclass `use X &&` wouldn't be needed, it's designed to do nothing if there's no .desktop and no icons needing cache updates.

true! good to know